.circleci/config.yml

```diff
@@ -243,6 +243,9 @@ jobs:
       - run:
           name: Run pytest
           working_directory: test_runner
+          # pytest doesn't output test logs in real time, so CI job may fail
+          # with `Too long with no output` error, if a test is running for a long time.
+          no_output_timeout: 20m
```

Comment:
It seems that each `run-pytest` completes in under ten minutes, and `test_restarts_under_load` in particular takes about three minutes only under normal circumstances: https://app.circleci.com/pipelines/github/zenithdb/zenith/3061/workflows/42b1e5ca-db66-41aa-9284-f9b3059ae060 .

I'm not sure if raising `no_output_timeout` to 20 minutes is a good idea. If a test understands that it's stuck or significantly slower than expected, it should fail itself. CI timelimits should be the last resort.

Reply:
I also prefer test fails to CI timeouts. I think raising `no_output_timeout` will help with that, at least temporarily.

Reply:
How about adding some code to `test_restarts_under_load` so it ensures a timeout of, say, 15 minutes on its own?

I think checking on each iteration is fine. 15 minutes + 200 seconds < 20 minutes.

Reply:
Just to note, there is handy plugin for pytest to enable per test timeouts https://pypi.org/project/pytest-timeout/.

Reply:
Good find. How dangerous is it? There are some flashy warnings in its README. For example, I'm not sure it will gracefully terminate the test and stop all processes, though the `signal` method looks promising. On the other hand, we use some C code, and if the signal handler gets called while GIL is locked... I don't know.

Reply:
I've used it in several python projects without problems. Timeout is indeed the last resort and in CI if there are any processes left they will die with the container anyway.